.github/workflows/reusable-change-detection.yml

```
@@ -127,12 +127,14 @@ jobs:
         format: csv  # works for paths with spaces
     - name: Check for docs changes
       if: >-
-        github.event_name == 'pull_request'
-        && steps.changed-docs-files.outputs.added_modified_renamed != ''
+        (github.event_name == 'pull_request'
+           && steps.changed-docs-files.outputs.added_modified_renamed != '')
+        || github.event_name == 'workflow_dispatch'
```

Nit: let's use the same formatting here and below. Either change this to match the MSI one:

```suggestion
        (github.event_name == 'pull_request'
           && steps.changed-docs-files.outputs.added_modified_renamed != ''
        ) || github.event_name == 'workflow_dispatch'
```